doc/logging/persistent_logger/persistent_loglevel.md

```diff
@@ -0,0 +1,460 @@
+
+# Motivation
+Log level verbosity is part of the configuration of the OS. Today, the log level is not persistent and gets a default value after reboot. It is required to add the ability to make the loglevel persistent.
+
+# Definitions/Abbreviation
+| Abbreviation  | Description                               |
+|---------------|-------------------------------------------|
+| SONiC         | Software for open networking in the cloud | 
+| SAI           | Switch Abstraction Interface              |
+| CONFIG DB	    | Configuration Database                    |
+
+
+# 1 Background
+Today, the user can configure the log level verbosity to each component in SWSS, Syncd, and SAI. In order to configure the log level in runtime, the user uses the **"swssloglevel" script**. The script updates the **LOGLEVEL DB** with the new verbosity. After a cold/fast reboot, the LOGLEVEL DB flushes, and the log level value of all components returns to default. For warm reboot, the log level will remain (as it is in today's implementation).
+
+## 1.1 "swssloglevel" usage
+
+The "swssloglevel" options:
+
+```
+admin@sonic:~$ swssloglevel -h
+Usage: swssloglevel [OPTIONS]
+SONiC logging severity level setting.
+
+Options:
+	 -h	print this message
+	 -l	loglevel value
+	 -c	component name in DB for which loglevel is applied (provided with -l)
+	 -a	apply loglevel to all components (provided with -l)
+	 -s	apply loglevel for SAI api component (equivalent to adding prefix "SAI_API_" to component)
+	 -p	print components registered in DB for which setting can be applied
+
+Examples:
+	swssloglevel -l NOTICE -c orchagent # set orchagent severity level to NOTICE
+	swssloglevel -l SAI_LOG_LEVEL_ERROR -s -c SWITCH # set SAI_API_SWITCH severity to ERROR
+	swssloglevel -l SAI_LOG_LEVEL_DEBUG -s -a # set all SAI_API_* severity to DEBUG
+
+```
+
+## 1.2 LOGLEVEL DB schema
+
+The LOGLEVEL DB is database #3. It contains a table for each component.
+Here is an example of the orchagent component table:
+
+```
+admin@sonic:~$ redis-cli -n 3 HGETALL "orchagent:orchagent"
+
+1) “LOGLEVEL”
+
+2) “INFO”
+
+3) “LOGOUTPUT”
+
+4) “SYSLOG”
+```
+
+
+# 2 Requirements Overview
+
+The persistent Logger should meet the following high-level functional requirement:
+- The user will be able to save the configuration of the loglevel and make it persistent after reboot.
+- Warm/Fast reboot won't be impacted following this change.
+
+# 3 Persistent log level design
+## 3.1 High-level design
+
+To make the loglevel persistent to reboot, we will move the Logger's tables in LOGLEVEL DB to CONFIG DB. Since the Config DB is already persistent, the log level will also be persistent to reboot. The log level will be saved using the "config save" CLI command.
+
+We will split this design into two phases to keep the pr with a small content:
+
+Phase 1: make the loglevel persistent by moving it to the CONFIG DB.
+
+Phase 2: Removing LOGLEVEL DB and the jinja2 cache.
```

Comment:
>jinja2 cache [](http://example.com/codeflow?start=38&length=12)

@stepanblyschak Please check whether it is possible to remove jinja2 cache.

Reply:
@qiluo-msft This cache was added to speed up template rendering in start scripts. There were a lot of them rendered during system start. This caused a delay in warm boot LAG restore time. According to the test did by @EdenGri with and without the cache we don't see any difference in this timing now. It is probably due to a lot of other optimizations done to sonic-cfggen. Since there is no noticible improvement made by j2 cache now it is safe to remove it.